升级到 3.2.10 后唤起键盘报错

试过缷载重新安装，清除rime目录重新添加部署都不行，3.2.9 正常。
下面是崩溃日志：
```
App Package Name: com.osfans.trime
App Version Name: 3.2.10
App Version Code: 20230101
OS Name: lineage_lemonade-userdebug 13 TQ1A.221205.011 9b431b5a42
OS Version: 5.4.219-qgki-g6c71020f9db9 (9b431b5a42)
OS API Level: 33
Device: OnePlus9
Model (product): LE2110 (OnePlus9)
Manufacturer: OnePlus
Tags: release-keys
Screen Size: 1080 x 2400
Screen Density: 3.0875
Screen orientation: Portrait
--------- beginning of main
I/libc    ( 8647): SetHeapTaggingLevel: tag level set to 0
--------- beginning of kernel
I/binder  ( 8647): 8647:8647 ioctl 40046210 7ffebaf1c0 returned -22
D/CompatibilityChangeReporter( 8647): Compat change id reported: 171979766; UID 10207; state: DISABLED
D/ApplicationLoaders( 8647): Returning zygote-cached class loader: /system/framework/android.test.base.jar
V/GraphicsEnvironment( 8647): ANGLE Developer option for 'com.osfans.trime' set to: 'default'
V/GraphicsEnvironment( 8647): ANGLE GameManagerService for com.osfans.trime: false
V/GraphicsEnvironment( 8647): Neither updatable production driver nor prerelease driver is supported.
D/NetworkSecurityConfig( 8647): No Network Security Config specified, using platform default
D/NetworkSecurityConfig( 8647): No Network Security Config specified, using platform default
I/CustomActivityOnCrash( 8647): CustomActivityOnCrash has been installed.
D/CompatibilityChangeReporter( 8647): Compat change id reported: 210923482; UID 10207; state: DISABLED
D/CompatibilityChangeReporter( 8647): Compat change id reported: 37756858; UID 10207; state: ENABLED
I/AdrenoGLES-0( 8647): QUALCOMM build                   : 5a9022f91f, Ib11adbd47c
I/AdrenoGLES-0( 8647): Build Date                       : 07/26/21
I/AdrenoGLES-0( 8647): OpenGL ES Shader Compiler Version: EV031.35.01.09
I/AdrenoGLES-0( 8647): Local Branch                     : 
I/AdrenoGLES-0( 8647): Remote Branch                    : 
I/AdrenoGLES-0( 8647): Remote Branch                    : 
I/AdrenoGLES-0( 8647): Reconstruct Branch               : 
I/AdrenoGLES-0( 8647): Build Config                     : S P 10.0.7 AArch64
I/AdrenoGLES-0( 8647): Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
I/AdrenoGLES-0( 8647): PFP: 0x016dc094, ME: 0x00000000
E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
I/Gralloc4( 8647): Adding additional valid usage bits: 0x8202000
W/Parcel  ( 8647): Expecting binder but got null!
D/CompatibilityChangeReporter( 8647): Compat change id reported: 160794467; UID 10207; state: DISABLED
E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
W/Parcel  ( 8647): Expecting binder but got null!
D/OpenGLRenderer( 8647): endAllActiveAnimators on 0x79a9e68060 (RippleDrawable) with handle 0x78b9e52240
E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
D/WindowTokenClient( 8647): Only apply configuration update to Resources because shouldReportConfigChange is false.
D/WindowTokenClient( 8647): android.window.WindowTokenClient.attachToDisplayArea:118 android.window.WindowContextController.attachToDisplayArea:107 android.window.WindowProviderService.attachBaseContext:161 android.app.Service.attach:915 android.app.ActivityThread.handleCreateService:4483 
I/rime.trime( 8647): registering core components.
W/om.osfans.trime( 8647): type=1400 audit(0.0:4435): avc: denied { create } for name="rime.trime.INFO" scontext=u:r:untrusted_app_29:s0:c207,c256,c512,c768 tcontext=u:object_r:fuse:s0:c207,c256,c512,c768 tclass=lnk_file permissive=0 app=com.osfans.trime
I/rime.trime( 8647): registering component: config_builder
I/rime.trime( 8647): registering component: config
I/rime.trime( 8647): registering component: schema
I/rime.trime( 8647): registering component: user_config
I/rime.trime( 8647): registering components from module 'dict'.
I/rime.trime( 8647): registering component: tabledb
I/rime.trime( 8647): registering component: stabledb
I/rime.trime( 8647): registering component: plain_userdb
I/rime.trime( 8647): registering component: userdb
I/rime.trime( 8647): registering component: corrector
I/rime.trime( 8647): registering component: dictionary
I/rime.trime( 8647): registering component: reverse_lookup_dictionary
I/rime.trime( 8647): registering component: user_dictionary
I/rime.trime( 8647): registering component: userdb_recovery_task
I/rime.trime( 8647): registering components from module 'gears'.
I/rime.trime( 8647): registering component: ascii_composer
I/rime.trime( 8647): registering component: chord_composer
I/rime.trime( 8647): registering component: express_editor
I/rime.trime( 8647): registering component: fluid_editor
I/rime.trime( 8647): registering component: fluency_editor
I/rime.trime( 8647): registering component: key_binder
I/rime.trime( 8647): registering component: navigator
I/rime.trime( 8647): registering component: punctuator
I/rime.trime( 8647): registering component: recognizer
I/rime.trime( 8647): registering component: selector
I/rime.trime( 8647): registering component: speller
I/rime.trime( 8647): registering component: shape_processor
I/rime.trime( 8647): registering component: abc_segmentor
I/rime.trime( 8647): registering component: affix_segmentor
I/rime.trime( 8647): registering component: ascii_segmentor
I/rime.trime( 8647): registering component: matcher
I/rime.trime( 8647): registering component: punct_segmentor
I/rime.trime( 8647): registering component: fallback_segmentor
I/rime.trime( 8647): registering component: echo_translator
I/rime.trime( 8647): registering component: punct_translator
I/rime.trime( 8647): registering component: table_translator
I/rime.trime( 8647): registering component: script_translator
I/rime.trime( 8647): registering component: r10n_translator
I/rime.trime( 8647): registering component: reverse_lookup_translator
I/rime.trime( 8647): registering component: schema_list_translator
I/rime.trime( 8647): registering component: switch_translator
I/rime.trime( 8647): registering component: history_translator
I/rime.trime( 8647): registering component: simplifier
I/rime.trime( 8647): registering component: uniquifier
I/rime.trime( 8647): registering component: charset_filter
I/rime.trime( 8647): registering component: cjk_minifier
I/rime.trime( 8647): registering component: reverse_lookup_filter
I/rime.trime( 8647): registering component: single_char_filter
I/rime.trime( 8647): registering component: shape_formatter
I/rime.trime( 8647): registering component: codepoint_translator
I/rime.trime( 8647): registering component: charset_filter
W/rime.trime( 8647): replacing previously registered component: charset_filter
I/rime.trime( 8647): registering components from module 'lua'.
W/om.osfans.trime( 8647): type=1400 audit(0.0:4436): avc: denied { create } for name="rime.trime.WARNING" scontext=u:r:untrusted_app_29:s0:c207,c256,c512,c768 tcontext=u:object_r:fuse:s0:c207,c256,c512,c768 tclass=lnk_file permissive=0 app=com.osfans.trime
I/rime.trime( 8647): rime.lua info: rime.lua should be either in the rime user data directory or in the rime shared data directory
I/rime.trime( 8647): registering component: lua_translator
I/rime.trime( 8647): registering component: lua_filter
I/rime.trime( 8647): registering component: lua_segmentor
I/rime.trime( 8647): registering component: lua_processor
I/rime.trime( 8647): registering components from module 'grammar'.
I/rime.trime( 8647): registering component: grammar
I/rime.trime( 8647): registering components from module 'levers'.
I/rime.trime( 8647): registering component: detect_modifications
I/rime.trime( 8647): registering component: installation_update
I/rime.trime( 8647): registering component: workspace_update
I/rime.trime( 8647): registering component: schema_update
I/rime.trime( 8647): registering component: config_file_update
I/rime.trime( 8647): registering component: prebuild_all_schemas
I/rime.trime( 8647): registering component: user_dict_upgrade
I/rime.trime( 8647): registering component: cleanup_trash
I/rime.trime( 8647): registering component: user_dict_sync
I/rime.trime( 8647): registering component: backup_config_files
I/rime.trime( 8647): registering component: clean_old_log_files
I/rime.trime( 8647): updating rime installation info.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/installation.yaml'.
I/rime.trime( 8647): installation info exists. installation id: 2334f55c-f015-4328-9c1b-8616932e7edc
I/rime.trime( 8647): sync dir: /storage/emulated/0/rime/sync
I/rime.trime( 8647): previous Rime version: 1.7.3
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 8647): modifications detected. workspace needs update.
I/rime.trime( 8647): changes detected; starting maintenance.
I/rime.trime( 8647): starting work thread for 3 tasks.
I/rime.trime( 8647): running deployment tasks:
I/rime.trime( 8647): updating workspace.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 8647): updating schemas.
I/rime.trime( 8647): schema: wubi98_mini
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/wubi98_mini.schema.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
I/rime.trime( 8647): preparing dictionary 'wubi98_mini'.
I/rime.trime( 8647): compiling dictionary for /storage/emulated/0/rime/build/wubi98_mini.schema.yaml
I/rime.trime( 8647): loading table file: /storage/emulated/0/rime/build/wubi98_mini.table.bin
I/rime.trime( 8647): loading prism file: /storage/emulated/0/rime/build/wubi98_mini.prism.bin
I/rime.trime( 8647): found double array image of size 56320.
I/rime.trime( 8647): "/storage/emulated/0/rime/wubi98_mini.dict.yaml"[1 file(s)] (1897802708)
I/rime.trime( 8647): /storage/emulated/0/rime/build/wubi98_mini.schema.yaml (3538624924)
I/rime.trime( 8647): loading reversedb: /storage/emulated/0/rime/build/wubi98_mini.reverse.bin
I/rime.trime( 8647): dictionary 'wubi98_mini' is ready.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
I/rime.trime( 8647): schema: pinyin_simp
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/pinyin_simp.schema.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/pinyin_simp.schema.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/pinyin_simp.schema.yaml'.
I/rime.trime( 8647): preparing dictionary 'pinyin_simp'.
I/rime.trime( 8647): compiling dictionary for /storage/emulated/0/rime/build/pinyin_simp.schema.yaml
I/rime.trime( 8647): loading table file: /storage/emulated/0/rime/build/pinyin_simp.table.bin
I/rime.trime( 8647): loading prism file: /storage/emulated/0/rime/build/pinyin_simp.prism.bin
I/rime.trime( 8647): found double array image of size 1536.
I/rime.trime( 8647): "/storage/emulated/0/rime/pinyin_simp.dict.yaml"[1 file(s)] (434301660)
I/rime.trime( 8647): /storage/emulated/0/rime/build/pinyin_simp.schema.yaml (971817021)
I/rime.trime( 8647): loading reversedb: /storage/emulated/0/rime/build/pinyin_simp.reverse.bin
I/rime.trime( 8647): dictionary 'pinyin_simp' is ready.
I/rime.trime( 8647): finished updating schemas: 2 success, 0 failure.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 8647): write: var/last_build_time
I/rime.trime( 8647): saving config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 8647): clean up trash.
I/rime.trime( 8647): 3 tasks ran: 3 success, 0 failure.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 8647): starting engine.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
I/rime.trime( 8647): updated option: _auto_commit
I/rime.trime( 8647): loading dictionary 'wubi98_mini'.
I/rime.trime( 8647): loading table file: /storage/emulated/0/rime/build/wubi98_mini.table.bin
I/rime.trime( 8647): loading prism file: /storage/emulated/0/rime/build/wubi98_mini.prism.bin
I/rime.trime( 8647): found double array image of size 56320.
I/rime.trime( 8647): updated option: ascii_mode
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/trime.yaml'.
I/rime.trime( 8647): source file changed: /storage/emulated/0/rime/trime.yaml
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/trime.yaml'.
I/rime.trime( 8647): auto-patch trime:/__patch: trime.custom:/patch?
W/rime.trime( 8647): circular dependencies detected in trime:
W/rime.trime( 8647): accessing blocking node with unresolved dependencies: trime:
I/rime.trime( 8647): resolved: Include(trime:/preset_keyboards/default)
I/rime.trime( 8647): resolved: PendingChild(trime:/preset_keyboards/letter)
W/rime.trime( 8647): circular dependencies detected in trime:
W/rime.trime( 8647): accessing blocking node with unresolved dependencies: trime:
I/rime.trime( 8647): resolved: Include(trime:/preset_keyboards/cangjie5)
I/rime.trime( 8647): resolved: PendingChild(trime:/preset_keyboards/scj6)
I/rime.trime( 8647): resolved: PendingChild(trime:/preset_keyboards)
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/trime.custom.yaml'.
I/rime.trime( 8647): patching preset_keyboards/wubi98_mini/import_preset
I/rime.trime( 8647): patching style/keyboards
I/rime.trime( 8647): resolved: Patch(trime.custom:patch <optional>)
I/rime.trime( 8647): saving config file '/storage/emulated/0/rime/build/trime.yaml'.
I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/trime.yaml'.
I/Choreographer( 8647): Skipped 39 frames!  The application may be doing too much work on its main thread.
D/WindowTokenClient( 8647): Configuration not dispatch to IME because configuration is up to date. Current config={1.0 460mcc7mnc [en_US] ldltr sw350dp w350dp h714dp 494dpi nrml long hdr widecg port finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 1080, 2400) mAppBounds=Rect(0, 123 - 1080, 2326) mMaxBounds=Rect(0, 0 - 1080, 2400) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.25 s.2086 fontWeightAdjustment=0}, reported config={1.0 460mcc7mnc [en_US] ldltr sw350dp w350dp h714dp 494dpi nrml long hdr widecg port finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 1080, 2400) mAppBounds=Rect(0, 123 - 1080, 2326) mMaxBounds=Rect(0, 0 - 1080, 2400) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.25 s.2086 fontWeightAdjustment=0}, updated config={1.0 460mcc7mnc [en_US] ldltr sw350dp w350dp h714dp 494dpi nrml long hdr widecg port finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 1080, 2400) mAppBounds=Rect(0, 123 - 1080, 2326) mMaxBounds=Rect(0, 0 - 1080, 2400) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.25 s.2086 fontWeightAdjustment=0}
D/CompatibilityChangeReporter( 8647): Compat change id reported: 78294732; UID 10207; state: DISABLED
D/CompatibilityChangeReporter( 8647): Compat change id reported: 147798919; UID 10207; state: DISABLED
E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
W/Parcel  ( 8647): Expecting binder but got null!
D/OpenGLRenderer( 8647): endAllActiveAnimators on 0x79a9e686d0 (RippleDrawable) with handle 0x78b9e84940
W/om.osfans.trime( 8647): Accessing hidden field Landroid/view/inputmethod/InputMethodManager;->mCurRootView:Landroid/view/ViewRootImpl; (blocked, reflection, denied)
D/AutofillManager( 8647): Fill dialog is enabled:false, hints=[]
D/CompatibilityChangeReporter( 8647): Compat change id reported: 171228096; UID 10207; state: ENABLED
E/ThemeUtils( 8647): View class com.osfans.trime.ime.text.Composition is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
I/System.out( 8647):  169 169 169 169
I/System.out( 8647):  187 187 187 190
W/om.osfans.trime( 8647): Cleared Reference was only reachable from finalizer (only reported once)
I/System.out( 8647):  135 135 135 135 135
I/System.out( 8647):  149 149 149 149 151
I/System.out( 8647):  135 135 135 135 135
I/System.out( 8647):  149 149 149 149 151
I/System.out( 8647):  135 135 135 135 135
I/System.out( 8647):  149 149 149 149 151
D/AndroidRuntime( 8647): Shutting down VM
--------- beginning of crash
E/AndroidRuntime( 8647): FATAL EXCEPTION: main
E/AndroidRuntime( 8647): Process: com.osfans.trime, PID: 8647
E/AndroidRuntime( 8647): java.lang.ExceptionInInitializerError
E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:32)
E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.core.Trime.onCreateInputView(Trime.java:733)
E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2088)
E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:2659)
E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:2588)
E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:928)
E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:902)
E/AndroidRuntime( 8647): 	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
E/AndroidRuntime( 8647): 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
E/AndroidRuntime( 8647): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime( 8647): 	at android.os.Looper.loopOnce(Looper.java:201)
E/AndroidRuntime( 8647): 	at android.os.Looper.loop(Looper.java:288)
E/AndroidRuntime( 8647): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
E/AndroidRuntime( 8647): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 8647): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/AndroidRuntime( 8647): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
E/AndroidRuntime( 8647): Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.Keyboard.<init>(Keyboard.java:242)
E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:37)
E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.<clinit>(KeyboardSwitcher.kt:27)
E/AndroidRuntime( 8647): 	... 16 more
E/CustomActivityOnCrash( 8647): App has crashed, executing CustomActivityOnCrash's UncaughtExceptionHandler
E/CustomActivityOnCrash( 8647): java.lang.ExceptionInInitializerError
E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:32)
E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.core.Trime.onCreateInputView(Trime.java:733)
E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2088)
E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:2659)
E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:2588)
E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:928)
E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:902)
E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
E/CustomActivityOnCrash( 8647): 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
E/CustomActivityOnCrash( 8647): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/CustomActivityOnCrash( 8647): 	at android.os.Looper.loopOnce(Looper.java:201)
E/CustomActivityOnCrash( 8647): 	at android.os.Looper.loop(Looper.java:288)
E/CustomActivityOnCrash( 8647): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
E/CustomActivityOnCrash( 8647): 	at java.lang.reflect.Method.invoke(Native Method)
E/CustomActivityOnCrash( 8647): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
E/CustomActivityOnCrash( 8647): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
E/CustomActivityOnCrash( 8647): Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.Keyboard.<init>(Keyboard.java:242)
E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:37)
E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.<clinit>(KeyboardSwitcher.kt:27)
E/CustomActivityOnCrash( 8647): 	... 16 more
E/CustomActivityOnCrash( 8647): Failed when resolving the restart activity class via getLaunchIntentForPackage, stack trace follows!
E/CustomActivityOnCrash( 8647): java.lang.ClassNotFoundException: com.osfans.trime.PrefLauncherAlias
E/CustomActivityOnCrash( 8647): 	at java.lang.Class.classForName(Native Method)
E/CustomActivityOnCrash( 8647): 	at java.lang.Class.forName(Class.java:454)
E/CustomActivityOnCrash( 8647): 	at java.lang.Class.forName(Class.java:379)
E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.getLauncherActivity(CustomActivityOnCrash.java:648)
E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.guessRestartActivityClass(CustomActivityOnCrash.java:602)
E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.lambda$install$0(CustomActivityOnCrash.java:177)
E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash$$ExternalSyntheticLambda0.uncaughtException(Unknown Source:2)
E/CustomActivityOnCrash( 8647): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1073)
E/CustomActivityOnCrash( 8647): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
E/CustomActivityOnCrash( 8647): 	at java.lang.Thread.dispatchUncaughtException(Thread.java:2306)
E/CustomActivityOnCrash( 8647): Caused by: java.lang.ClassNotFoundException: com.osfans.trime.PrefLauncherAlias
E/CustomActivityOnCrash( 8647): 	... 10 more
I/Process ( 8647): Sending signal. PID: 8647 SIG: 9
```

我这里安装3.2.10后也出现程序崩溃情况。
<code>
App Package Name: com.osfans.trime
App Version Name: 3.2.10
App Version Code: 20230101
OS Name: QKQ1.190828.002 test-keys
OS Version: 4.9.186-perf-g10af704 (V12.0.5.0.QECCNXM)
OS API Level: 29
Device: equuleus
Model (product): MI 8 UD (equuleus)
Manufacturer: Xiaomi
Tags: release-keys
Screen Size: 1080 x 2248
Screen Density: 2.75
Screen orientation: Portrait
--------- beginning of main
E/om.osfans.trim( 9926): Not starting debugger since process cannot load the jdwp agent.
I/Perf    ( 9926): Connecting to perf service.
I/CustomActivityOnCrash( 9926): CustomActivityOnCrash has been installed.
I/FeatureParser( 9926): can't find equuleus.xml in assets/device_features/,it may be in /system/etc/device_features
E/libc    ( 9926): Access denied finding property "ro.vendor.df.effect.conflict"
E/Perf    ( 9926): Fail to get file list com.osfans.trime
E/Perf    ( 9926): getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
E/Perf    ( 9926): Fail to get file list oat
E/Perf    ( 9926): getFolderSize() : Exception_1 = java.lang.NullPointerException: Attempt to get length of null array
I/TextToSpeech( 9926): Sucessfully bound to com.github.olga_yakovleva.rhvoice.android
I/rime.trime( 9926): registering core components.
I/rime.trime( 9926): registering component: config_builder
I/rime.trime( 9926): registering component: config
I/rime.trime( 9926): registering component: schema
I/rime.trime( 9926): registering component: user_config
I/rime.trime( 9926): registering components from module 'dict'.
I/rime.trime( 9926): registering component: tabledb
I/rime.trime( 9926): registering component: stabledb
I/rime.trime( 9926): registering component: plain_userdb
I/rime.trime( 9926): registering component: userdb
I/rime.trime( 9926): registering component: corrector
I/rime.trime( 9926): registering component: dictionary
I/rime.trime( 9926): registering component: reverse_lookup_dictionary
I/rime.trime( 9926): registering component: user_dictionary
I/rime.trime( 9926): registering component: userdb_recovery_task
I/rime.trime( 9926): registering components from module 'gears'.
I/rime.trime( 9926): registering component: ascii_composer
I/rime.trime( 9926): registering component: chord_composer
I/rime.trime( 9926): registering component: express_editor
I/rime.trime( 9926): registering component: fluid_editor
I/rime.trime( 9926): registering component: fluency_editor
I/rime.trime( 9926): registering component: key_binder
I/rime.trime( 9926): registering component: navigator
I/rime.trime( 9926): registering component: punctuator
I/rime.trime( 9926): registering component: recognizer
I/rime.trime( 9926): registering component: selector
I/rime.trime( 9926): registering component: speller
I/rime.trime( 9926): registering component: shape_processor
I/rime.trime( 9926): registering component: abc_segmentor
I/rime.trime( 9926): registering component: affix_segmentor
I/rime.trime( 9926): registering component: ascii_segmentor
I/rime.trime( 9926): registering component: matcher
I/rime.trime( 9926): registering component: punct_segmentor
I/rime.trime( 9926): registering component: fallback_segmentor
I/rime.trime( 9926): registering component: echo_translator
I/rime.trime( 9926): registering component: punct_translator
I/rime.trime( 9926): registering component: table_translator
I/rime.trime( 9926): registering component: script_translator
I/rime.trime( 9926): registering component: r10n_translator
I/rime.trime( 9926): registering component: reverse_lookup_translator
I/rime.trime( 9926): registering component: schema_list_translator
I/rime.trime( 9926): registering component: switch_translator
I/rime.trime( 9926): registering component: history_translator
I/rime.trime( 9926): registering component: simplifier
I/rime.trime( 9926): registering component: uniquifier
I/rime.trime( 9926): registering component: charset_filter
I/rime.trime( 9926): registering component: cjk_minifier
I/rime.trime( 9926): registering component: reverse_lookup_filter
I/rime.trime( 9926): registering component: single_char_filter
I/rime.trime( 9926): registering component: shape_formatter
I/rime.trime( 9926): registering component: codepoint_translator
I/rime.trime( 9926): registering component: charset_filter
W/rime.trime( 9926): replacing previously registered component: charset_filter
I/rime.trime( 9926): registering components from module 'lua'.
I/rime.trime( 9926): rime.lua info: rime.lua should be either in the rime user data directory or in the rime shared data directory
I/rime.trime( 9926): registering component: lua_translator
I/rime.trime( 9926): registering component: lua_filter
I/rime.trime( 9926): registering component: lua_segmentor
I/rime.trime( 9926): registering component: lua_processor
I/rime.trime( 9926): registering components from module 'grammar'.
I/rime.trime( 9926): registering component: grammar
I/rime.trime( 9926): registering components from module 'levers'.
I/rime.trime( 9926): registering component: detect_modifications
I/rime.trime( 9926): registering component: installation_update
I/rime.trime( 9926): registering component: workspace_update
I/rime.trime( 9926): registering component: schema_update
I/rime.trime( 9926): registering component: config_file_update
I/rime.trime( 9926): registering component: prebuild_all_schemas
I/rime.trime( 9926): registering component: user_dict_upgrade
I/rime.trime( 9926): registering component: cleanup_trash
I/rime.trime( 9926): registering component: user_dict_sync
I/rime.trime( 9926): registering component: backup_config_files
I/rime.trime( 9926): registering component: clean_old_log_files
I/rime.trime( 9926): updating rime installation info.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/installation.yaml'.
I/rime.trime( 9926): installation info exists. installation id: 4098908e-50ad-45e6-bd3e-0725f0c5faa0
I/rime.trime( 9926): sync dir: /storage/emulated/0/rime/sync
I/rime.trime( 9926): previous Rime version: 1.7.3
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 9926): modifications detected. workspace needs update.
I/rime.trime( 9926): changes detected; starting maintenance.
I/rime.trime( 9926): starting work thread for 3 tasks.
I/rime.trime( 9926): running deployment tasks:
I/rime.trime( 9926): updating workspace.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 9926): updating schemas.
I/rime.trime( 9926): schema: luna_pinyin_simp
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): preparing dictionary 'luna_pinyin'.
I/rime.trime( 9926): compiling dictionary for /storage/emulated/0/rime/build/luna_pinyin_simp.schema.yaml
I/rime.trime( 9926): loading table file: /storage/emulated/0/rime/build/luna_pinyin.table.bin
I/rime.trime( 9926): loading prism file: /storage/emulated/0/rime/build/luna_pinyin_simp.prism.bin
I/rime.trime( 9926): found double array image of size 1792.
I/rime.trime( 9926): "/storage/emulated/0/rime/luna_pinyin.dict.yaml"[1 file(s)] (630016651)
I/rime.trime( 9926): /storage/emulated/0/rime/build/luna_pinyin_simp.schema.yaml (2113915002)
I/rime.trime( 9926): loading reversedb: /storage/emulated/0/rime/build/luna_pinyin.reverse.bin
I/rime.trime( 9926): dictionary 'luna_pinyin' is ready.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): finished updating schemas: 1 success, 0 failure.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 9926): write: var/last_build_time
I/rime.trime( 9926): saving config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 9926): clean up trash.
I/rime.trime( 9926): 3 tasks ran: 3 success, 0 failure.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 9926): starting engine.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/user.yaml'.
I/rime.trime( 9926): updated option: ascii_punct
I/rime.trime( 9926): updated option: full_shape
I/rime.trime( 9926): updated option: simplification
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): updated option: _auto_commit
I/rime.trime( 9926): stabledb 'custom_phrase' does not exist.
I/rime.trime( 9926): loading dictionary 'luna_pinyin'.
I/rime.trime( 9926): loading table file: /storage/emulated/0/rime/build/luna_pinyin.table.bin
I/rime.trime( 9926): loading prism file: /storage/emulated/0/rime/build/luna_pinyin_simp.prism.bin
I/rime.trime( 9926): found double array image of size 1792.
I/rime.trime( 9926): updated option: ascii_mode
I/rime.trime( 9926): updated option: zh_simp
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/trime.yaml'.
I/rime.trime( 9926): source file changed: /storage/emulated/0/rime/trime.yaml
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/trime.yaml'.
I/rime.trime( 9926): auto-patch trime:/__patch: trime.custom:/patch?
W/rime.trime( 9926): circular dependencies detected in trime:
W/rime.trime( 9926): accessing blocking node with unresolved dependencies: trime:
I/rime.trime( 9926): resolved: Include(trime:/preset_keyboards/default)
I/rime.trime( 9926): resolved: PendingChild(trime:/preset_keyboards/letter)
W/rime.trime( 9926): circular dependencies detected in trime:
W/rime.trime( 9926): accessing blocking node with unresolved dependencies: trime:
I/rime.trime( 9926): resolved: Include(trime:/preset_keyboards/cangjie5)
I/rime.trime( 9926): resolved: PendingChild(trime:/preset_keyboards/scj6)
I/rime.trime( 9926): resolved: PendingChild(trime:/preset_keyboards)
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/trime.custom.yaml'.
W/rime.trime( 9926): circular dependencies detected in trime:
W/rime.trime( 9926): accessing blocking node with unresolved dependencies: trime:
I/rime.trime( 9926): resolved: Include(trime:/preset_keys)
I/rime.trime( 9926): resolved: PendingChild(trime.custom:/patch/preset_keys)
I/rime.trime( 9926): patching liquid_keyboard/emoji/keys
I/rime.trime( 9926): patching liquid_keyboard/keyboards
I/rime.trime( 9926): patching liquid_keyboard/symbol/keys
I/rime.trime( 9926): patching liquid_keyboard/symbol/type
I/rime.trime( 9926): patching preset_color_schemes/default/pwhite
I/rime.trime( 9926): patching preset_keyboards/luna_pinyin/ascii_mode
I/rime.trime( 9926): patching preset_keyboards/luna_pinyin_simp/ascii_mode
I/rime.trime( 9926): patching preset_keyboards/luna_pinyin_simp/import_preset
I/rime.trime( 9926): patching preset_keyboards/qwertz
I/rime.trime( 9926): patching preset_keyboards/yxsymbols
I/rime.trime( 9926): patching preset_keys
I/rime.trime( 9926): patching preset_keys/Keyboard_yxsymbols
I/rime.trime( 9926): patching style/horizontal_gap
I/rime.trime( 9926): patching style/keyboards
I/rime.trime( 9926): patching style/reset_ascii_mode
I/rime.trime( 9926): patching style/vertical_gap
I/rime.trime( 9926): resolved: Patch(trime.custom:patch <optional>)
I/rime.trime( 9926): saving config file '/storage/emulated/0/rime/build/trime.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/trime.yaml'.
I/TextToSpeech( 9926): Connected to ComponentInfo{com.github.olga_yakovleva.rhvoice.android/com.github.olga_yakovleva.rhvoice.android.RHVoiceService}
I/TextToSpeech( 9926): Set up connection to ComponentInfo{com.github.olga_yakovleva.rhvoice.android/com.github.olga_yakovleva.rhvoice.android.RHVoiceService}
E/ThemeUtils( 9926): View class com.osfans.trime.ime.text.Composition is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
I/System.out( 9926):  151 151 151 151
I/System.out( 9926):  144 144 144 145
I/System.out( 9926):  123 123 123 123
I/System.out( 9926):  139 139 139 140
I/System.out( 9926):  151 151 151 151
I/System.out( 9926):  144 144 144 145
I/System.out( 9926):  123 123 123 123
I/System.out( 9926):  139 139 139 140
I/System.out( 9926):  151 151 151 151
I/System.out( 9926):  144 144 144 145
I/System.out( 9926):  123 123 123 123
I/System.out( 9926):  139 139 139 140
I/AdrenoGLES( 9926): QUALCOMM build                   : e541a88, I20154638fb
I/AdrenoGLES( 9926): Build Date                       : 09/15/20
I/AdrenoGLES( 9926): OpenGL ES Shader Compiler Version: EV031.27.05.01
I/AdrenoGLES( 9926): Local Branch                     : 
I/AdrenoGLES( 9926): Remote Branch                    : refs/tags/AU_LINUX_ANDROID_LA.UM.8.3.R1.10.00.00.520.058
I/AdrenoGLES( 9926): Remote Branch                    : NONE
I/AdrenoGLES( 9926): Reconstruct Branch               : NOTHING
I/AdrenoGLES( 9926): Build Config                     : S P 8.0.11 AArch64
W/RenderThread( 9926): type=1400 audit(0.0:5753298): avc: denied { search } for name="clients" dev="debugfs" ino=1139 scontext=u:r:untrusted_app:s0:c27,c260,c512,c768 tcontext=u:object_r:debugfs_ion:s0 tclass=dir permissive=0
I/AdrenoGLES( 9926): PFP: 0x016ee187, ME: 0x00000000
W/AdrenoUtils( 9926): <ReadGpuID_from_sysfs:194>: Failed to open /sys/class/kgsl/kgsl-3d0/gpu_model
W/AdrenoUtils( 9926): <ReadGpuID:218>: Failed to read chip ID from gpu_model. Fallback to use the GSL path
W/Gralloc3( 9926): mapper 3.x is not supported
I/rime.trime( 9926): updated option: soft_cursor
I/rime.trime( 9926): updated option: _horizontal
D/OpenGLRenderer( 9926): endAllActiveAnimators on 0x7c95fbd800 (RippleDrawable) with handle 0x7cee730880
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/default.custom.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/bopomofo_tw.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/terra_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/bopomofo_tw.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/terra_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/build/default.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/default.custom.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/bopomofo_tw.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/terra_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin_simp.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/bopomofo_tw.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/terra_pinyin.schema.yaml'.
I/rime.trime( 9926): loading config file '/storage/emulated/0/rime/luna_pinyin_simp.schema.yaml'.
D/OpenGLRenderer( 9926): endAllActiveAnimators on 0x7cfb1cdf00 (RippleDrawable) with handle 0x7cee736380
--------- beginning of system
I/Timeline( 9926): Timeline: Activity_launch_request time:168126440
D/OpenGLRenderer( 9926): endAllActiveAnimators on 0x7cfaab1c00 (RippleDrawable) with handle 0x7cee736c20
W/ActivityThread( 9926): handleWindowVisibility: no activity for token android.os.BinderProxy@bafe3dd
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ViewContentFactory( 9926): initViewContentFetcherClass
D/ViewContentFactory( 9926): getInterceptorPackageInfo
D/ViewContentFactory( 9926): getInitialApplication took 1ms
D/ViewContentFactory( 9926): packageInfo.packageName: com.miui.catcherpatch
D/ViewContentFactory( 9926): initViewContentFetcherClass took 9ms
I/ContentCatcher( 9926): ViewContentFetcher : ViewContentFetcher
D/ViewContentFactory( 9926): createInterceptor took 9ms
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
I/chatty  ( 9926): uid=11051(com.osfans.trime) identical 1 line
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
D/ForceDarkHelper( 9926): updateByCheckExcludeList: pkg: com.osfans.trime activity: com.osfans.trime.ui.main.PrefMainActivity@dfcca03
W/RenderInspector( 9926): DequeueBuffer time out on com.osfans.trime/com.osfans.trime.ui.main.PrefMainActivity, count=2, avg=20 ms, max=21 ms.
W/om.osfans.trim( 9926): Accessing hidden field Landroid/view/inputmethod/InputMethodManager;->mServedView:Landroid/view/View; (greylist-max-p, reflection, denied)
W/om.osfans.trim( 9926): Accessing hidden field Landroid/view/inputmethod/InputMethodManager;->mNextServedView:Landroid/view/View; (greylist-max-p, reflection, denied)
E/InputEventReceiver( 9926): Exception dispatching input event.
E/MessageQueue-JNI( 9926): Exception in MessageQueue callback: handleReceiveCallback
E/MessageQueue-JNI( 9926): java.lang.ArrayIndexOutOfBoundsException: length=2; index=-1
E/MessageQueue-JNI( 9926): 	at java.util.ArrayList.get(ArrayList.java:439)
E/MessageQueue-JNI( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.switchKeyboard(KeyboardSwitcher.kt:77)
E/MessageQueue-JNI( 9926): 	at com.osfans.trime.ime.text.TextInputManager.onEvent(TextInputManager.kt:354)
E/MessageQueue-JNI( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.detectAndSendKey(KeyboardView.java:1110)
E/MessageQueue-JNI( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.onModifiedTouchEvent(KeyboardView.java:1699)
E/MessageQueue-JNI( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.onTouchEvent(KeyboardView.java:1512)
E/MessageQueue-JNI( 9926): 	at android.view.View.dispatchTouchEvent(View.java:13484)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/MessageQueue-JNI( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/MessageQueue-JNI( 9926): 	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:498)
E/MessageQueue-JNI( 9926): 	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1853)
E/MessageQueue-JNI( 9926): 	at android.app.Dialog.dispatchTouchEvent(Dialog.java:863)
E/MessageQueue-JNI( 9926): 	at android.inputmethodservice.SoftInputWindow.dispatchTouchEvent(SoftInputWindow.java:161)
E/MessageQueue-JNI( 9926): 	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:456)
E/MessageQueue-JNI( 9926): 	at android.view.View.dispatchPointerEvent(View.java:13745)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5637)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5437)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4991)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4957)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5097)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4965)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5154)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4991)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4957)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4965)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7690)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7659)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7620)
E/MessageQueue-JNI( 9926): 	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7820)
E/MessageQueue-JNI( 9926): 	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:251)
E/MessageQueue-JNI( 9926): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MessageQueue-JNI( 9926): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/MessageQueue-JNI( 9926): 	at android.os.Looper.loop(Looper.java:181)
E/MessageQueue-JNI( 9926): 	at android.app.ActivityThread.main(ActivityThread.java:7562)
E/MessageQueue-JNI( 9926): 	at java.lang.reflect.Method.invoke(Native Method)
E/MessageQueue-JNI( 9926): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
E/MessageQueue-JNI( 9926): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
D/AndroidRuntime( 9926): Shutting down VM
--------- beginning of crash
E/AndroidRuntime( 9926): FATAL EXCEPTION: main
E/AndroidRuntime( 9926): Process: com.osfans.trime, PID: 9926
E/AndroidRuntime( 9926): java.lang.ArrayIndexOutOfBoundsException: length=2; index=-1
E/AndroidRuntime( 9926): 	at java.util.ArrayList.get(ArrayList.java:439)
E/AndroidRuntime( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.switchKeyboard(KeyboardSwitcher.kt:77)
E/AndroidRuntime( 9926): 	at com.osfans.trime.ime.text.TextInputManager.onEvent(TextInputManager.kt:354)
E/AndroidRuntime( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.detectAndSendKey(KeyboardView.java:1110)
E/AndroidRuntime( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.onModifiedTouchEvent(KeyboardView.java:1699)
E/AndroidRuntime( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.onTouchEvent(KeyboardView.java:1512)
E/AndroidRuntime( 9926): 	at android.view.View.dispatchTouchEvent(View.java:13484)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/AndroidRuntime( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/AndroidRuntime( 9926): 	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:498)
E/AndroidRuntime( 9926): 	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1853)
E/AndroidRuntime( 9926): 	at android.app.Dialog.dispatchTouchEvent(Dialog.java:863)
E/AndroidRuntime( 9926): 	at android.inputmethodservice.SoftInputWindow.dispatchTouchEvent(SoftInputWindow.java:161)
E/AndroidRuntime( 9926): 	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:456)
E/AndroidRuntime( 9926): 	at android.view.View.dispatchPointerEvent(View.java:13745)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5637)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5437)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4991)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4957)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5097)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4965)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5154)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4991)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4957)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4965)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7690)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7659)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7620)
E/AndroidRuntime( 9926): 	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7820)
E/AndroidRuntime( 9926): 	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:251)
E/AndroidRuntime( 9926): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/AndroidRuntime( 9926): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/AndroidRuntime( 9926): 	at android.os.Looper.loop(Looper.java:181)
E/AndroidRuntime( 9926): 	at android.app.ActivityThread.main(ActivityThread.java:7562)
E/AndroidRuntime( 9926): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 9926): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
E/AndroidRuntime( 9926): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
E/CustomActivityOnCrash( 9926): App has crashed, executing CustomActivityOnCrash's UncaughtExceptionHandler
E/CustomActivityOnCrash( 9926): java.lang.ArrayIndexOutOfBoundsException: length=2; index=-1
E/CustomActivityOnCrash( 9926): 	at java.util.ArrayList.get(ArrayList.java:439)
E/CustomActivityOnCrash( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.switchKeyboard(KeyboardSwitcher.kt:77)
E/CustomActivityOnCrash( 9926): 	at com.osfans.trime.ime.text.TextInputManager.onEvent(TextInputManager.kt:354)
E/CustomActivityOnCrash( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.detectAndSendKey(KeyboardView.java:1110)
E/CustomActivityOnCrash( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.onModifiedTouchEvent(KeyboardView.java:1699)
E/CustomActivityOnCrash( 9926): 	at com.osfans.trime.ime.keyboard.KeyboardView.onTouchEvent(KeyboardView.java:1512)
E/CustomActivityOnCrash( 9926): 	at android.view.View.dispatchTouchEvent(View.java:13484)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:3088)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2781)
E/CustomActivityOnCrash( 9926): 	at com.android.internal.policy.DecorView.superDispatchTouchEvent(DecorView.java:498)
E/CustomActivityOnCrash( 9926): 	at com.android.internal.policy.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1853)
E/CustomActivityOnCrash( 9926): 	at android.app.Dialog.dispatchTouchEvent(Dialog.java:863)
E/CustomActivityOnCrash( 9926): 	at android.inputmethodservice.SoftInputWindow.dispatchTouchEvent(SoftInputWindow.java:161)
E/CustomActivityOnCrash( 9926): 	at com.android.internal.policy.DecorView.dispatchTouchEvent(DecorView.java:456)
E/CustomActivityOnCrash( 9926): 	at android.view.View.dispatchPointerEvent(View.java:13745)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:5637)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:5437)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4991)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4957)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:5097)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4965)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:5154)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4991)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4957)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4965)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4938)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:7690)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:7659)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:7620)
E/CustomActivityOnCrash( 9926): 	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:7820)
E/CustomActivityOnCrash( 9926): 	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:251)
E/CustomActivityOnCrash( 9926): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/CustomActivityOnCrash( 9926): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/CustomActivityOnCrash( 9926): 	at android.os.Looper.loop(Looper.java:181)
E/CustomActivityOnCrash( 9926): 	at android.app.ActivityThread.main(ActivityThread.java:7562)
E/CustomActivityOnCrash( 9926): 	at java.lang.reflect.Method.invoke(Native Method)
E/CustomActivityOnCrash( 9926): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
E/CustomActivityOnCrash( 9926): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
E/CustomActivityOnCrash( 9926): Failed when resolving the restart activity class via getLaunchIntentForPackage, stack trace follows!
E/CustomActivityOnCrash( 9926): java.lang.ClassNotFoundException: com.osfans.trime.PrefLauncherAlias
E/CustomActivityOnCrash( 9926): 	at java.lang.Class.classForName(Native Method)
E/CustomActivityOnCrash( 9926): 	at java.lang.Class.forName(Class.java:454)
E/CustomActivityOnCrash( 9926): 	at java.lang.Class.forName(Class.java:379)
E/CustomActivityOnCrash( 9926): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.getLauncherActivity(CustomActivityOnCrash.java:648)
E/CustomActivityOnCrash( 9926): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.guessRestartActivityClass(CustomActivityOnCrash.java:602)
E/CustomActivityOnCrash( 9926): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.lambda$install$0(CustomActivityOnCrash.java:177)
E/CustomActivityOnCrash( 9926): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash$$ExternalSyntheticLambda0.uncaughtException(Unknown Source:2)
E/CustomActivityOnCrash( 9926): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1073)
E/CustomActivityOnCrash( 9926): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
E/CustomActivityOnCrash( 9926): 	at java.lang.Thread.dispatchUncaughtException(Thread.java:2187)
E/CustomActivityOnCrash( 9926): Caused by: java.lang.ClassNotFoundException: com.osfans.trime.PrefLauncherAlias
E/CustomActivityOnCrash( 9926): 	... 10 more
I/Timeline( 9926): Timeline: Activity_launch_request time:168133361
I/Process ( 9926): Sending signal. PID: 9926 SIG: 9
</code>

> 试过缷载重新安装，清除rime目录重新添加部署都不行，3.2.9 正常。 下面是崩溃日志：
> 
> ```
> App Package Name: com.osfans.trime
> App Version Name: 3.2.10
> App Version Code: 20230101
> OS Name: lineage_lemonade-userdebug 13 TQ1A.221205.011 9b431b5a42
> OS Version: 5.4.219-qgki-g6c71020f9db9 (9b431b5a42)
> OS API Level: 33
> Device: OnePlus9
> Model (product): LE2110 (OnePlus9)
> Manufacturer: OnePlus
> Tags: release-keys
> Screen Size: 1080 x 2400
> Screen Density: 3.0875
> Screen orientation: Portrait
> --------- beginning of main
> I/libc    ( 8647): SetHeapTaggingLevel: tag level set to 0
> --------- beginning of kernel
> I/binder  ( 8647): 8647:8647 ioctl 40046210 7ffebaf1c0 returned -22
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 171979766; UID 10207; state: DISABLED
> D/ApplicationLoaders( 8647): Returning zygote-cached class loader: /system/framework/android.test.base.jar
> V/GraphicsEnvironment( 8647): ANGLE Developer option for 'com.osfans.trime' set to: 'default'
> V/GraphicsEnvironment( 8647): ANGLE GameManagerService for com.osfans.trime: false
> V/GraphicsEnvironment( 8647): Neither updatable production driver nor prerelease driver is supported.
> D/NetworkSecurityConfig( 8647): No Network Security Config specified, using platform default
> D/NetworkSecurityConfig( 8647): No Network Security Config specified, using platform default
> I/CustomActivityOnCrash( 8647): CustomActivityOnCrash has been installed.
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 210923482; UID 10207; state: DISABLED
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 37756858; UID 10207; state: ENABLED
> I/AdrenoGLES-0( 8647): QUALCOMM build                   : 5a9022f91f, Ib11adbd47c
> I/AdrenoGLES-0( 8647): Build Date                       : 07/26/21
> I/AdrenoGLES-0( 8647): OpenGL ES Shader Compiler Version: EV031.35.01.09
> I/AdrenoGLES-0( 8647): Local Branch                     : 
> I/AdrenoGLES-0( 8647): Remote Branch                    : 
> I/AdrenoGLES-0( 8647): Remote Branch                    : 
> I/AdrenoGLES-0( 8647): Reconstruct Branch               : 
> I/AdrenoGLES-0( 8647): Build Config                     : S P 10.0.7 AArch64
> I/AdrenoGLES-0( 8647): Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
> I/AdrenoGLES-0( 8647): PFP: 0x016dc094, ME: 0x00000000
> E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
> I/Gralloc4( 8647): Adding additional valid usage bits: 0x8202000
> W/Parcel  ( 8647): Expecting binder but got null!
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 160794467; UID 10207; state: DISABLED
> E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
> W/Parcel  ( 8647): Expecting binder but got null!
> D/OpenGLRenderer( 8647): endAllActiveAnimators on 0x79a9e68060 (RippleDrawable) with handle 0x78b9e52240
> E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
> D/WindowTokenClient( 8647): Only apply configuration update to Resources because shouldReportConfigChange is false.
> D/WindowTokenClient( 8647): android.window.WindowTokenClient.attachToDisplayArea:118 android.window.WindowContextController.attachToDisplayArea:107 android.window.WindowProviderService.attachBaseContext:161 android.app.Service.attach:915 android.app.ActivityThread.handleCreateService:4483 
> I/rime.trime( 8647): registering core components.
> W/om.osfans.trime( 8647): type=1400 audit(0.0:4435): avc: denied { create } for name="rime.trime.INFO" scontext=u:r:untrusted_app_29:s0:c207,c256,c512,c768 tcontext=u:object_r:fuse:s0:c207,c256,c512,c768 tclass=lnk_file permissive=0 app=com.osfans.trime
> I/rime.trime( 8647): registering component: config_builder
> I/rime.trime( 8647): registering component: config
> I/rime.trime( 8647): registering component: schema
> I/rime.trime( 8647): registering component: user_config
> I/rime.trime( 8647): registering components from module 'dict'.
> I/rime.trime( 8647): registering component: tabledb
> I/rime.trime( 8647): registering component: stabledb
> I/rime.trime( 8647): registering component: plain_userdb
> I/rime.trime( 8647): registering component: userdb
> I/rime.trime( 8647): registering component: corrector
> I/rime.trime( 8647): registering component: dictionary
> I/rime.trime( 8647): registering component: reverse_lookup_dictionary
> I/rime.trime( 8647): registering component: user_dictionary
> I/rime.trime( 8647): registering component: userdb_recovery_task
> I/rime.trime( 8647): registering components from module 'gears'.
> I/rime.trime( 8647): registering component: ascii_composer
> I/rime.trime( 8647): registering component: chord_composer
> I/rime.trime( 8647): registering component: express_editor
> I/rime.trime( 8647): registering component: fluid_editor
> I/rime.trime( 8647): registering component: fluency_editor
> I/rime.trime( 8647): registering component: key_binder
> I/rime.trime( 8647): registering component: navigator
> I/rime.trime( 8647): registering component: punctuator
> I/rime.trime( 8647): registering component: recognizer
> I/rime.trime( 8647): registering component: selector
> I/rime.trime( 8647): registering component: speller
> I/rime.trime( 8647): registering component: shape_processor
> I/rime.trime( 8647): registering component: abc_segmentor
> I/rime.trime( 8647): registering component: affix_segmentor
> I/rime.trime( 8647): registering component: ascii_segmentor
> I/rime.trime( 8647): registering component: matcher
> I/rime.trime( 8647): registering component: punct_segmentor
> I/rime.trime( 8647): registering component: fallback_segmentor
> I/rime.trime( 8647): registering component: echo_translator
> I/rime.trime( 8647): registering component: punct_translator
> I/rime.trime( 8647): registering component: table_translator
> I/rime.trime( 8647): registering component: script_translator
> I/rime.trime( 8647): registering component: r10n_translator
> I/rime.trime( 8647): registering component: reverse_lookup_translator
> I/rime.trime( 8647): registering component: schema_list_translator
> I/rime.trime( 8647): registering component: switch_translator
> I/rime.trime( 8647): registering component: history_translator
> I/rime.trime( 8647): registering component: simplifier
> I/rime.trime( 8647): registering component: uniquifier
> I/rime.trime( 8647): registering component: charset_filter
> I/rime.trime( 8647): registering component: cjk_minifier
> I/rime.trime( 8647): registering component: reverse_lookup_filter
> I/rime.trime( 8647): registering component: single_char_filter
> I/rime.trime( 8647): registering component: shape_formatter
> I/rime.trime( 8647): registering component: codepoint_translator
> I/rime.trime( 8647): registering component: charset_filter
> W/rime.trime( 8647): replacing previously registered component: charset_filter
> I/rime.trime( 8647): registering components from module 'lua'.
> W/om.osfans.trime( 8647): type=1400 audit(0.0:4436): avc: denied { create } for name="rime.trime.WARNING" scontext=u:r:untrusted_app_29:s0:c207,c256,c512,c768 tcontext=u:object_r:fuse:s0:c207,c256,c512,c768 tclass=lnk_file permissive=0 app=com.osfans.trime
> I/rime.trime( 8647): rime.lua info: rime.lua should be either in the rime user data directory or in the rime shared data directory
> I/rime.trime( 8647): registering component: lua_translator
> I/rime.trime( 8647): registering component: lua_filter
> I/rime.trime( 8647): registering component: lua_segmentor
> I/rime.trime( 8647): registering component: lua_processor
> I/rime.trime( 8647): registering components from module 'grammar'.
> I/rime.trime( 8647): registering component: grammar
> I/rime.trime( 8647): registering components from module 'levers'.
> I/rime.trime( 8647): registering component: detect_modifications
> I/rime.trime( 8647): registering component: installation_update
> I/rime.trime( 8647): registering component: workspace_update
> I/rime.trime( 8647): registering component: schema_update
> I/rime.trime( 8647): registering component: config_file_update
> I/rime.trime( 8647): registering component: prebuild_all_schemas
> I/rime.trime( 8647): registering component: user_dict_upgrade
> I/rime.trime( 8647): registering component: cleanup_trash
> I/rime.trime( 8647): registering component: user_dict_sync
> I/rime.trime( 8647): registering component: backup_config_files
> I/rime.trime( 8647): registering component: clean_old_log_files
> I/rime.trime( 8647): updating rime installation info.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/installation.yaml'.
> I/rime.trime( 8647): installation info exists. installation id: 2334f55c-f015-4328-9c1b-8616932e7edc
> I/rime.trime( 8647): sync dir: /storage/emulated/0/rime/sync
> I/rime.trime( 8647): previous Rime version: 1.7.3
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/user.yaml'.
> I/rime.trime( 8647): modifications detected. workspace needs update.
> I/rime.trime( 8647): changes detected; starting maintenance.
> I/rime.trime( 8647): starting work thread for 3 tasks.
> I/rime.trime( 8647): running deployment tasks:
> I/rime.trime( 8647): updating workspace.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/default.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/default.yaml'.
> I/rime.trime( 8647): updating schemas.
> I/rime.trime( 8647): schema: wubi98_mini
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/wubi98_mini.schema.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
> I/rime.trime( 8647): preparing dictionary 'wubi98_mini'.
> I/rime.trime( 8647): compiling dictionary for /storage/emulated/0/rime/build/wubi98_mini.schema.yaml
> I/rime.trime( 8647): loading table file: /storage/emulated/0/rime/build/wubi98_mini.table.bin
> I/rime.trime( 8647): loading prism file: /storage/emulated/0/rime/build/wubi98_mini.prism.bin
> I/rime.trime( 8647): found double array image of size 56320.
> I/rime.trime( 8647): "/storage/emulated/0/rime/wubi98_mini.dict.yaml"[1 file(s)] (1897802708)
> I/rime.trime( 8647): /storage/emulated/0/rime/build/wubi98_mini.schema.yaml (3538624924)
> I/rime.trime( 8647): loading reversedb: /storage/emulated/0/rime/build/wubi98_mini.reverse.bin
> I/rime.trime( 8647): dictionary 'wubi98_mini' is ready.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
> I/rime.trime( 8647): schema: pinyin_simp
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/pinyin_simp.schema.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/pinyin_simp.schema.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/pinyin_simp.schema.yaml'.
> I/rime.trime( 8647): preparing dictionary 'pinyin_simp'.
> I/rime.trime( 8647): compiling dictionary for /storage/emulated/0/rime/build/pinyin_simp.schema.yaml
> I/rime.trime( 8647): loading table file: /storage/emulated/0/rime/build/pinyin_simp.table.bin
> I/rime.trime( 8647): loading prism file: /storage/emulated/0/rime/build/pinyin_simp.prism.bin
> I/rime.trime( 8647): found double array image of size 1536.
> I/rime.trime( 8647): "/storage/emulated/0/rime/pinyin_simp.dict.yaml"[1 file(s)] (434301660)
> I/rime.trime( 8647): /storage/emulated/0/rime/build/pinyin_simp.schema.yaml (971817021)
> I/rime.trime( 8647): loading reversedb: /storage/emulated/0/rime/build/pinyin_simp.reverse.bin
> I/rime.trime( 8647): dictionary 'pinyin_simp' is ready.
> I/rime.trime( 8647): finished updating schemas: 2 success, 0 failure.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/user.yaml'.
> I/rime.trime( 8647): write: var/last_build_time
> I/rime.trime( 8647): saving config file '/storage/emulated/0/rime/user.yaml'.
> I/rime.trime( 8647): clean up trash.
> I/rime.trime( 8647): 3 tasks ran: 3 success, 0 failure.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/default.yaml'.
> I/rime.trime( 8647): starting engine.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/user.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/wubi98_mini.schema.yaml'.
> I/rime.trime( 8647): updated option: _auto_commit
> I/rime.trime( 8647): loading dictionary 'wubi98_mini'.
> I/rime.trime( 8647): loading table file: /storage/emulated/0/rime/build/wubi98_mini.table.bin
> I/rime.trime( 8647): loading prism file: /storage/emulated/0/rime/build/wubi98_mini.prism.bin
> I/rime.trime( 8647): found double array image of size 56320.
> I/rime.trime( 8647): updated option: ascii_mode
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/trime.yaml'.
> I/rime.trime( 8647): source file changed: /storage/emulated/0/rime/trime.yaml
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/trime.yaml'.
> I/rime.trime( 8647): auto-patch trime:/__patch: trime.custom:/patch?
> W/rime.trime( 8647): circular dependencies detected in trime:
> W/rime.trime( 8647): accessing blocking node with unresolved dependencies: trime:
> I/rime.trime( 8647): resolved: Include(trime:/preset_keyboards/default)
> I/rime.trime( 8647): resolved: PendingChild(trime:/preset_keyboards/letter)
> W/rime.trime( 8647): circular dependencies detected in trime:
> W/rime.trime( 8647): accessing blocking node with unresolved dependencies: trime:
> I/rime.trime( 8647): resolved: Include(trime:/preset_keyboards/cangjie5)
> I/rime.trime( 8647): resolved: PendingChild(trime:/preset_keyboards/scj6)
> I/rime.trime( 8647): resolved: PendingChild(trime:/preset_keyboards)
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/trime.custom.yaml'.
> I/rime.trime( 8647): patching preset_keyboards/wubi98_mini/import_preset
> I/rime.trime( 8647): patching style/keyboards
> I/rime.trime( 8647): resolved: Patch(trime.custom:patch <optional>)
> I/rime.trime( 8647): saving config file '/storage/emulated/0/rime/build/trime.yaml'.
> I/rime.trime( 8647): loading config file '/storage/emulated/0/rime/build/trime.yaml'.
> I/Choreographer( 8647): Skipped 39 frames!  The application may be doing too much work on its main thread.
> D/WindowTokenClient( 8647): Configuration not dispatch to IME because configuration is up to date. Current config={1.0 460mcc7mnc [en_US] ldltr sw350dp w350dp h714dp 494dpi nrml long hdr widecg port finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 1080, 2400) mAppBounds=Rect(0, 123 - 1080, 2326) mMaxBounds=Rect(0, 0 - 1080, 2400) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.25 s.2086 fontWeightAdjustment=0}, reported config={1.0 460mcc7mnc [en_US] ldltr sw350dp w350dp h714dp 494dpi nrml long hdr widecg port finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 1080, 2400) mAppBounds=Rect(0, 123 - 1080, 2326) mMaxBounds=Rect(0, 0 - 1080, 2400) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.25 s.2086 fontWeightAdjustment=0}, updated config={1.0 460mcc7mnc [en_US] ldltr sw350dp w350dp h714dp 494dpi nrml long hdr widecg port finger -keyb/v/h -nav/h winConfig={ mBounds=Rect(0, 0 - 1080, 2400) mAppBounds=Rect(0, 123 - 1080, 2326) mMaxBounds=Rect(0, 0 - 1080, 2400) mDisplayRotation=ROTATION_0 mWindowingMode=fullscreen mDisplayWindowingMode=fullscreen mActivityType=undefined mAlwaysOnTop=undefined mRotation=ROTATION_0} as.25 s.2086 fontWeightAdjustment=0}
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 78294732; UID 10207; state: DISABLED
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 147798919; UID 10207; state: DISABLED
> E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
> E/OpenGLRenderer( 8647): Unable to match the desired swap behavior.
> W/Parcel  ( 8647): Expecting binder but got null!
> D/OpenGLRenderer( 8647): endAllActiveAnimators on 0x79a9e686d0 (RippleDrawable) with handle 0x78b9e84940
> W/om.osfans.trime( 8647): Accessing hidden field Landroid/view/inputmethod/InputMethodManager;->mCurRootView:Landroid/view/ViewRootImpl; (blocked, reflection, denied)
> D/AutofillManager( 8647): Fill dialog is enabled:false, hints=[]
> D/CompatibilityChangeReporter( 8647): Compat change id reported: 171228096; UID 10207; state: ENABLED
> E/ThemeUtils( 8647): View class com.osfans.trime.ime.text.Composition is an AppCompat widget that can only be used with a Theme.AppCompat theme (or descendant).
> I/System.out( 8647):  169 169 169 169
> I/System.out( 8647):  187 187 187 190
> W/om.osfans.trime( 8647): Cleared Reference was only reachable from finalizer (only reported once)
> I/System.out( 8647):  135 135 135 135 135
> I/System.out( 8647):  149 149 149 149 151
> I/System.out( 8647):  135 135 135 135 135
> I/System.out( 8647):  149 149 149 149 151
> I/System.out( 8647):  135 135 135 135 135
> I/System.out( 8647):  149 149 149 149 151
> D/AndroidRuntime( 8647): Shutting down VM
> --------- beginning of crash
> E/AndroidRuntime( 8647): FATAL EXCEPTION: main
> E/AndroidRuntime( 8647): Process: com.osfans.trime, PID: 8647
> E/AndroidRuntime( 8647): java.lang.ExceptionInInitializerError
> E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:32)
> E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.core.Trime.onCreateInputView(Trime.java:733)
> E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2088)
> E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:2659)
> E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:2588)
> E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:928)
> E/AndroidRuntime( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:902)
> E/AndroidRuntime( 8647): 	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
> E/AndroidRuntime( 8647): 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
> E/AndroidRuntime( 8647): 	at android.os.Handler.dispatchMessage(Handler.java:106)
> E/AndroidRuntime( 8647): 	at android.os.Looper.loopOnce(Looper.java:201)
> E/AndroidRuntime( 8647): 	at android.os.Looper.loop(Looper.java:288)
> E/AndroidRuntime( 8647): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
> E/AndroidRuntime( 8647): 	at java.lang.reflect.Method.invoke(Native Method)
> E/AndroidRuntime( 8647): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
> E/AndroidRuntime( 8647): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
> E/AndroidRuntime( 8647): Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
> E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.Keyboard.<init>(Keyboard.java:242)
> E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:37)
> E/AndroidRuntime( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.<clinit>(KeyboardSwitcher.kt:27)
> E/AndroidRuntime( 8647): 	... 16 more
> E/CustomActivityOnCrash( 8647): App has crashed, executing CustomActivityOnCrash's UncaughtExceptionHandler
> E/CustomActivityOnCrash( 8647): java.lang.ExceptionInInitializerError
> E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:32)
> E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.core.Trime.onCreateInputView(Trime.java:733)
> E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService.updateInputViewShown(InputMethodService.java:2088)
> E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService.prepareWindow(InputMethodService.java:2659)
> E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService.showWindow(InputMethodService.java:2588)
> E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInput(InputMethodService.java:928)
> E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.InputMethodService$InputMethodImpl.showSoftInputWithToken(InputMethodService.java:902)
> E/CustomActivityOnCrash( 8647): 	at android.inputmethodservice.IInputMethodWrapper.executeMessage(IInputMethodWrapper.java:232)
> E/CustomActivityOnCrash( 8647): 	at com.android.internal.os.HandlerCaller$MyHandler.handleMessage(HandlerCaller.java:44)
> E/CustomActivityOnCrash( 8647): 	at android.os.Handler.dispatchMessage(Handler.java:106)
> E/CustomActivityOnCrash( 8647): 	at android.os.Looper.loopOnce(Looper.java:201)
> E/CustomActivityOnCrash( 8647): 	at android.os.Looper.loop(Looper.java:288)
> E/CustomActivityOnCrash( 8647): 	at android.app.ActivityThread.main(ActivityThread.java:7872)
> E/CustomActivityOnCrash( 8647): 	at java.lang.reflect.Method.invoke(Native Method)
> E/CustomActivityOnCrash( 8647): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
> E/CustomActivityOnCrash( 8647): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
> E/CustomActivityOnCrash( 8647): Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.util.Iterator java.util.List.iterator()' on a null object reference
> E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.Keyboard.<init>(Keyboard.java:242)
> E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.newOrReset(KeyboardSwitcher.kt:37)
> E/CustomActivityOnCrash( 8647): 	at com.osfans.trime.ime.keyboard.KeyboardSwitcher.<clinit>(KeyboardSwitcher.kt:27)
> E/CustomActivityOnCrash( 8647): 	... 16 more
> E/CustomActivityOnCrash( 8647): Failed when resolving the restart activity class via getLaunchIntentForPackage, stack trace follows!
> E/CustomActivityOnCrash( 8647): java.lang.ClassNotFoundException: com.osfans.trime.PrefLauncherAlias
> E/CustomActivityOnCrash( 8647): 	at java.lang.Class.classForName(Native Method)
> E/CustomActivityOnCrash( 8647): 	at java.lang.Class.forName(Class.java:454)
> E/CustomActivityOnCrash( 8647): 	at java.lang.Class.forName(Class.java:379)
> E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.getLauncherActivity(CustomActivityOnCrash.java:648)
> E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.guessRestartActivityClass(CustomActivityOnCrash.java:602)
> E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash.lambda$install$0(CustomActivityOnCrash.java:177)
> E/CustomActivityOnCrash( 8647): 	at cat.ereza.customactivityoncrash.CustomActivityOnCrash$$ExternalSyntheticLambda0.uncaughtException(Unknown Source:2)
> E/CustomActivityOnCrash( 8647): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1073)
> E/CustomActivityOnCrash( 8647): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
> E/CustomActivityOnCrash( 8647): 	at java.lang.Thread.dispatchUncaughtException(Thread.java:2306)
> E/CustomActivityOnCrash( 8647): Caused by: java.lang.ClassNotFoundException: com.osfans.trime.PrefLauncherAlias
> E/CustomActivityOnCrash( 8647): 	... 10 more
> I/Process ( 8647): Sending signal. PID: 8647 SIG: 9
> ```

你尝试一下最新的版本,
我这现在已经可以正常弹出键盘了,之前也是不行

--- a/app/src/main/java/com/osfans/trime/data/theme/Theme.kt
+++ b/app/src/main/java/com/osfans/trime/data/theme/Theme.kt
@@ -250,9 +250,8 @@ class Theme {
             val remapped =
                 if (".default" == name) {
                     val currentSchemaId = Rime.getCurrentRimeSchema()
-                    val shortSchemaId = currentSchemaId.split("_".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0]
-                    if (theme.presetKeyboards!!.containsKey(shortSchemaId)) {
-                        return shortSchemaId
+                    if (theme.presetKeyboards!!.containsKey(currentSchemaId)) {
+                        return currentSchemaId
                     } else {
                         val alphabet = SchemaManager.getActiveSchema().alphabet
                         val twentySix = "qwerty"

--- a/app/src/main/java/com/osfans/trime/ime/core/Trime.java
+++ b/app/src/main/java/com/osfans/trime/ime/core/Trime.java
@@ -966,7 +966,13 @@ public class Trime extends LifecycleInputMethodService {
   @Override
   public boolean onKeyDown(int keyCode, KeyEvent event) {
     Timber.i("\t<TrimeInput>\tonKeyDown()\tkeycode=%d, event=%s", keyCode, event.toString());
-    if (composeEvent(event) && onKeyEvent(event)) return true;
+    if (composeEvent(event) && onKeyEvent(event)) {
+      if (!isWindowShown) {
+        return super.onKeyDown(keyCode, event);
+      } else {
+        return true;
+      }
+    }
     return super.onKeyDown(keyCode, event);
   }
 
@@ -975,7 +981,11 @@ public class Trime extends LifecycleInputMethodService {
     Timber.i("\t<TrimeInput>\tonKeyUp()\tkeycode=%d, event=%s", keyCode, event.toString());
     if (composeEvent(event) && textInputManager.getNeedSendUpRimeKey()) {
       textInputManager.onRelease(keyCode);
-      return true;
+      if (!isWindowShown) {
+        return super.onKeyUp(keyCode, event);
+      } else {
+        return true;
+      }
     }
     return super.onKeyUp(keyCode, event);
   }

